upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -197,3 +199,30 @@ func (tf *TemplateFunctions) ExternalDnsArgv() ([]string, error) {
 
 	return argv, nil
 }
+
+func (tf *TemplateFunctions) ProxyEnv() (map[string]string, error) {
+	envs := map[string]string{}
+	proxies := tf.cluster.Spec.EgressProxy
+	if proxies == nil {
+		return envs, nil
```

Comment:
EgressProxy is optional, having it be nil is not an error to me, returning an empty map means this function could be composed with others that return environment variables.